tests/unit_tests/lr_warmup_test.py

```diff
@@ -110,6 +110,47 @@ def test_lr_warmup_with_lr_scheduling(self):
         # THE LRS AFTER THE UPDATE
         self.assertListEqual(lrs, expected_lrs)
 
+    def test_lr_warmup_per_step_with_lr_scheduling(self):
+        # Define model
+        net = LeNet()
+        trainer = Trainer("lr_warmup_test_per_step")
+
+        lrs = []
+        phase_callbacks = [TestLRCallback(lr_placeholder=lrs)]
```

Comment:
I think it would be better to have another callback that would collect the lrs after batch_step (or alternatively at batch start) so we know the warmup actually happened.